File: 2023/15/15.py
total = 0

def _hash(_s: str):
    h: int = 0
    for i in _s:
        h = (h + ord(i)) * 17 % 256
    return h

for step in open("15.txt").readline().strip().split(','):
    v = 0
    v = _hash(step)
    total += v
print(total)

File: 2023/15/15.scala
import scala.io.Source


def hash (s: String): Int {
    var h: Int = 0
    for (i < s){
        h = (h + i.toInt()) * 17 % 256
    }
    return h
}

object Main extends App {
  var total = 0

  for (step <- Source.fromFile("15.txt").getLines().next().split(',')) {
    var v = 0
    v = hash(step)
    total += v
  }

  println(total)
}